src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClientFactory.cs

```diff
@@ -76,19 +76,12 @@ public IDurableClient CreateClient(DurableClientOptions durableClientOptions)
 
             DurableClientAttribute attribute = new DurableClientAttribute(durableClientOptions);
 
-            HttpApiHandler httpApiHandler = this.cachedHttpListeners.GetOrAdd(
-                attribute,
-                attr =>
-                {
-                    return new HttpApiHandler(null, null, this.durableTaskOptions, this.logger);
-                });
-
             DurableClient client = this.cachedClients.GetOrAdd(
                 attribute,
                 attr =>
                 {
                     DurabilityProvider innerClient = this.durabilityProviderFactory.GetDurabilityProvider(attribute);
-                    return new DurableClient(innerClient, httpApiHandler, attribute, this.MessageDataConverter, this.TraceHelper, this.durableTaskOptions);
+                    return new DurableClient(innerClient, null, attribute, this.MessageDataConverter, this.TraceHelper, this.durableTaskOptions);
```

Comment:
I believe this factory is also the implementation of  `IDurableClientFactory` for DurableTaskExtension, so I don't think we can safely set `httpApiHandler` to null here always. 

Is there a way to safely distinguish when we are running in a `DurableTaskExtension` and we can safey provide a `HttpApiHandler` instance, and when we can just specify `null`?

Reply:
Can you point to where `DurableTaskExtension` is using `IDurableClientFactory`?

`DurableClientAttribute.IsExternalClient` could be a way to check if httpApiHandler should be set to null or not.

Reply:
Woops, I am mistaken! I assumed it got injected via dependency injection in DurableTaskExtension, but it looks like that is not the case? 

I am a bit confused of the need for `DurableClientAttrbiute.ExternalClient` then, but if I remember correctly that may be a remnant from some work @amdeel did a few releases back, not this new DurableClientFactory work...